Incorrect Lifetime Learning Credit amount in years before 2021

Before 2021, the AGI phase-out parameters for the Lifetime Learning Credit (calculated in Part II of IRS Form 8863) differed from the AGI phase-out parameters for the American Opportunity Credit (calculated in Part I of IRS Form 8863).  In 2021, these two sets of phase-out parameters were the same, and that is how OpenFisca-US 0.136.0 version has it for **all** years.  Because the LLC phase-out begins at lower AGI than the AOC phase-out begin in years before 2021, the current code in OpenFisca-US sometimes assigns a positive LLC amount to a tax unit that in reality has AGI too large to receive a LLC.

The parameter differences in years before 2021 can be read straight off each Form 8863, links to which are provided on this [IRS webpage](https://apps.irs.gov/app/picklist/list/priorFormPublication.html?resultsPerPage=200&sortColumn=sortOrder&indexOfFirstRow=0&criteria=formNumber&value=8863&isDescending=false).

The following 2020 test case illustrates this problem.  First, the YAML test file, then the test results, and finally excerpts from the OpenFisca-US test trace and the USI-Tax-Analyzer dump output.  The expected income tax in the test is derived from the USI-Tax-Analyzer because TAXSIM35 does not simulate education credits.

Here is the test file:
```
- name: Tax unit created by hand to illustrate pre-2021 LLC problem
  absolute_error_margin: 0.01
  period: 2020
  input:
    people:
      person1:
        is_tax_unit_head: 1
        is_tax_unit_spouse: 0
        is_tax_unit_dependent: 0
        age: 25
        ssi: 0
        state_supplement: 0
        wic: 0
        qualified_tuition_expenses: 4000
        is_eligible_for_american_opportunity_credit: 0
      person2:
        is_tax_unit_head: 0
        is_tax_unit_spouse: 1
        is_tax_unit_dependent: 0
        age: 25
        ssi: 0
        state_supplement: 0
        wic: 0
        employment_income: 150000
    spm_units:
      spm_unit:
        members: [person1, person2]
        snap: 0
        tanf: 0
    tax_units:
      tax_unit:
        members: [person1, person2]
        premium_tax_credit: 0
    households:
      household:
        members: [person1, person2]
        state_code: TX
  output:
    income_tax: 14644
```

Here are the test results:
```
================================= FAILURES ===================================
_______________________________ test session _________________________________
/Users/mrh/work/l20id1.test.yaml:
  Test 'Tax unit created by hand to illustrate pre-2021 LLC problem':
    income_tax@2020: [13844.] differs from 14644.0
      with an absolute margin [800.] > 0.01
========================= short test summary info ============================
FAILED l20id1.test.yaml::
============================ 1 failed in 0.64s ===============================
```

Here are some excerpts from the OpenFisca-US test trace:
```
adjusted_gross_income<2020> >> [146000.]
lifetime_learning_credit<2020> >> [800.]
recovery_rebate_credit<2020> >> [3600.]
income_tax<2020> >> [13844.]
```

And here are some excerpts from the USI-Tax-Analyzer dump output:
```
c00100 146000.0   <--- AGI
c07230 0.0        <--- LLC
rrc 3600.0
itax 14644.0  
```

So, the $800 income tax difference is caused by the fact that USI-Tax-Analyzer is using the AGI phase-out parameters specified by IRS on the 2020 Form 8863 while OpenFisca-US is neglecting to do this in 2020 and years before 2020.

--- a/changelog.d/fix-issue-1324.fixed.md
+++ b/changelog.d/fix-issue-1324.fixed.md
@@ -1,0 +1,1 @@
+Corrected the Lifetime Learning Credit phase-out thresholds to use pre-2021 IRS Form 8863 values, which differed from the American Opportunity Credit before the Consolidated Appropriations Act, 2021 harmonized them.

--- a/policyengine_us/parameters/gov/irs/credits/education/lifetime_learning_credit/phase_out.yaml
+++ b/policyengine_us/parameters/gov/irs/credits/education/lifetime_learning_credit/phase_out.yaml
@@ -1,0 +1,67 @@
+start:
+  description: Phase-out start for the Lifetime Learning Credit on AGI. Before 2021, the LLC used lower thresholds than the American Opportunity Credit; the Consolidated Appropriations Act, 2021 harmonized them (IRC 25A(i)).
+  single:
+    description: Phase-out start for the Lifetime Learning Credit on AGI (single filers).
+    values:
+      2010-01-01: 50_000
+      2011-01-01: 51_000
+      2012-01-01: 52_000
+      2013-01-01: 53_000
+      2014-01-01: 54_000
+      2015-01-01: 55_000
+      2017-01-01: 56_000
+      2018-01-01: 57_000
+      2019-01-01: 58_000
+      2020-01-01: 59_000
+      2021-01-01: 80_000
+    metadata:
+      unit: currency-USD
+      period: year
+      label: Lifetime Learning Credit phase-out start (single filers)
+  joint:
+    description: Phase-out start for the Lifetime Learning Credit on AGI (joint filers).
+    values:
+      2010-01-01: 100_000
+      2011-01-01: 102_000
+      2012-01-01: 104_000
+      2013-01-01: 107_000
+      2014-01-01: 108_000
+      2015-01-01: 110_000
+      2016-01-01: 111_000
+      2017-01-01: 112_000
+      2018-01-01: 114_000
+      2019-01-01: 116_000
+      2020-01-01: 118_000
+      2021-01-01: 160_000
+    metadata:
+      unit: currency-USD
+      period: year
+      label: Lifetime Learning Credit phase-out start (joint filers)
+length:
+  description: Length of the phase-out range for the Lifetime Learning Credit on AGI (excess income at which the credit is reduced to zero).
+  single:
+    description: Length of the phase-out range for the Lifetime Learning Credit on AGI (single filers).
+    values:
+      2010-01-01: 10_000
+    metadata:
+      unit: currency-USD
+      period: year
+      label: Lifetime Learning Credit phase-out length (single filers)
+  joint:
+    description: Length of the phase-out range for the Lifetime Learning Credit on AGI (joint filers).
+    values:
+      2010-01-01: 20_000
+    metadata:
+      unit: currency-USD
+      period: year
+      label: Lifetime Learning Credit phase-out length (joint filers)
+metadata:
+  reference:
+    - title: 26 U.S. Code § 25A(d)(1)
+      href: https://www.law.cornell.edu/uscode/text/26/25A#d_1
+    - title: 2020 IRS Form 8863 (Part II lines 13, 16)
+      href: https://www.irs.gov/pub/irs-prior/f8863--2020.pdf
+    - title: 2015 IRS Form 8863 (Part II lines 13, 16)
+      href: https://www.irs.gov/pub/irs-prior/f8863--2015.pdf
+    - title: IRC 25A(i) (harmonized phase-out from 2021)
+      href: https://www.law.cornell.edu/uscode/text/26/25A#i

--- a/policyengine_us/tests/policy/baseline/gov/irs/credits/education/lifetime_learning_credit/lifetime_learning_credit_pre_2021_phase_out.yaml
+++ b/policyengine_us/tests/policy/baseline/gov/irs/credits/education/lifetime_learning_credit/lifetime_learning_credit_pre_2021_phase_out.yaml
@@ -1,0 +1,54 @@
+# Regression tests for issue #1324.
+# Before 2021, the Lifetime Learning Credit used separate AGI phase-out
+# parameters from the American Opportunity Credit (see IRS Form 8863 Part II
+# lines 13/16). The LLC phase-out began at lower AGI than the AOC until the
+# Consolidated Appropriations Act, 2021 harmonized them at $80k single /
+# $160k joint (IRC 25A(i)).
+#
+# Prior to the fix, both credits shared the AOC phase-out, so LLC was
+# computed as nonzero for filers who in fact had AGI above the LLC phase-out.
+
+- name: 2020 joint filer with AGI $146k is above the $138k LLC phase-out end (issue #1324)
+  period: 2020
+  input:
+    adjusted_gross_income: 146_000
+    qualified_tuition_expenses: 4_000
+    filing_status: JOINT
+    tax_unit_is_joint: true
+    is_eligible_for_american_opportunity_credit: false
+  output:
+    # Joint LLC phase-out in 2020: starts $118k, ends $138k.
+    # AGI $146k is above phase-out end, so LLC potential is $0.
+    # Before the fix, code used AOC joint params ($160k/$180k), so LLC
+    # potential was computed as $800 - 800 * (146000 - 160000)/20000 clamped
+    # = $800 (since AGI was below the AOC phase-out start).
+    lifetime_learning_credit_potential: 0
+
+- name: 2020 single filer with AGI $65k is partly phased out for the LLC (issue #1324)
+  period: 2020
+  input:
+    adjusted_gross_income: 65_000
+    qualified_tuition_expenses: 5_000
+    filing_status: SINGLE
+    tax_unit_is_joint: false
+    is_eligible_for_american_opportunity_credit: false
+  output:
+    # Single LLC phase-out in 2020: starts $59k, ends $69k.
+    # Potential (pre-phase-out) = 0.2 * min(5000, 10000) = $1,000.
+    # Phase-out fraction = (65000 - 59000) / 10000 = 0.6.
+    # Potential = 1000 * (1 - 0.6) = $400.
+    lifetime_learning_credit_potential: 400
+
+- name: 2021 joint filer uses harmonized $160k/$180k phase-out
+  period: 2021
+  input:
+    adjusted_gross_income: 146_000
+    qualified_tuition_expenses: 4_000
+    filing_status: JOINT
+    tax_unit_is_joint: true
+    is_eligible_for_american_opportunity_credit: false
+  output:
+    # Starting 2021, LLC and AOC share phase-out ($160k/$180k joint).
+    # AGI $146k is below the $160k start: no phase-out applied.
+    # Potential = 0.2 * min(4000, 10000) = $800.
+    lifetime_learning_credit_potential: 800

--- a/policyengine_us/variables/gov/irs/credits/education/lifetime_learning_credit_phase_out.py
+++ b/policyengine_us/variables/gov/irs/credits/education/lifetime_learning_credit_phase_out.py
@@ -1,0 +1,34 @@
+from policyengine_us.model_api import *
+
+
+class lifetime_learning_credit_phase_out(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Lifetime Learning Credit phase-out"
+    unit = "/1"
+    documentation = (
+        "Fraction of the Lifetime Learning Credit that is phased out "
+        "based on AGI. Before 2021 the LLC had lower phase-out thresholds "
+        "than the American Opportunity Credit (IRS Form 8863, Part II); "
+        "the Consolidated Appropriations Act, 2021 harmonized them "
+        "(IRC 25A(i))."
+    )
+    definition_period = YEAR
+    reference = "https://www.law.cornell.edu/uscode/text/26/25A#d_1"
+
+    def formula(tax_unit, period, parameters):
+        llc = parameters(period).gov.irs.credits.education.lifetime_learning_credit
+        agi = tax_unit("adjusted_gross_income", period)
+        is_joint = tax_unit("tax_unit_is_joint", period)
+        phase_out_start = where(
+            is_joint,
+            llc.phase_out.start.joint,
+            llc.phase_out.start.single,
+        )
+        phase_out_length = where(
+            is_joint,
+            llc.phase_out.length.joint,
+            llc.phase_out.length.single,
+        )
+        excess_agi = max_(0, agi - phase_out_start)
+        return min_(1, excess_agi / phase_out_length)

--- a/policyengine_us/variables/gov/irs/credits/education/lifetime_learning_credit_potential.py
+++ b/policyengine_us/variables/gov/irs/credits/education/lifetime_learning_credit_potential.py
@@ -20,7 +20,7 @@ class lifetime_learning_credit_potential(Variable):
         )
         capped_expenses = min_(llc.expense_limit, eligible_expenses)
         maximum_amount = llc.rate * capped_expenses
-        phase_out = tax_unit("education_credit_phase_out", period)
+        phase_out = tax_unit("lifetime_learning_credit_phase_out", period)
         if llc.abolition:
             return 0
         return max_(0, maximum_amount * (1 - phase_out))

